pkg/js/validator/validate-dsl.ts

```diff
@@ -0,0 +1,14 @@
+import { OpenFgaDslSyntaxMultipleError, parseDSL } from "../transformer/dsltojson";
+
+/**
+ * validateDSL - Validates model authored in FGA DSL syntax, returning all found errors
+ * @param {string} dsl 
+ * @returns {OpenFgaDslSyntaxMultipleError}
+ */
+export default function validateDsl(dsl: string): OpenFgaDslSyntaxMultipleError {
+    const { errorListener } = parseDSL(dsl);
+  
+    return new OpenFgaDslSyntaxMultipleError(errorListener.errors);
```

Comment:
I'm not sure about this - shouldn't this be throwing errors, and returning null if the model is valid?

Here you are always returning an error.

Reply:
Happy to talk about this one. My reason;
* If you go throw error, return null - you need to try catch to collect failures. If i return an array of errors, its just a functional call.
* On the case with no errors, `errorListener.errors` is an empty array, so if you parse correctly the function returns that empty array.

Either way will work - I just leant this way to avoid try-catching for every invocation.

Reply:
If we are staying with the return error/nil method, maybe we should change the wording of [validate-dsl.test.ts](https://github.com/openfga/language/pull/34/files#diff-44c1dbcbe9b8294cd670c05737a815d5df7dc41587993cc486a518e622928a1a) as we are not throwing error.

Reply:
Yep, can do that.

Do you have any strong opinions supporting either method?

Reply:
I definitely do not like always returning an error, it must be a nil if it is valid. You should not get an error that has an empty array for errors

Reply:
And I personally prefer that we throw. It makes more sense to me. When we validate, we listen to the error and show it to the user, and if an error is not thrown, we proceed normally

Reply:
Cool - fair reasoning, not going to argue. I'll make the change in favor of the throw.